src/sirocco/workgraph.py

```diff
@@ -317,6 +322,26 @@ def _set_shelljob_arguments(self, task: core.ShellTask):
         _, arguments = self.split_cmd_arg(task.resolve_ports(input_labels))
         workgraph_task_arguments.value = arguments
 
+    def _set_shelljob_filenames(self, task: core.ShellTask):
+        """set AiiDA ShellJob filenames for AvailableData entities"""
+        workgraph_task = self.task_from_core(task)
+
+        filenames = {}
+        input_sockets = workgraph_task.inputs.nodes
```

Comment:
The current implementation is a bit hacky and there already are helper functions for accessing the aiida nodes from the core nodes.
```suggestion
        for input_ in task.inputs:
            if isinstance(input_, core.AvailableData):
                input_socket = self.data_from_core(input_)
                filenames[input_socket._name] = Path(input_.src).name
```